### DISCUSSION TITLE
How to Obtain Conversation UUID for Custom CSAT Survey URL

### DISCUSSION THREAD

**Discussion by MichelN89** [chatwoot/chatwoot#9885] [Q&A] [answered]
<img width="1198" alt="image" src="https://github.com/user-attachments/assets/4d4535e5-7b13-49ad-96ce-0c9658127ade">

Hello everyone!

I'm working on setting up a custom flow for the CSAT survey and need some assistance. I'm sending the emails from my own server based on several conditions and want to include a URL to the survey page in that email. According to the API documentation, the URL requires a {conversation_uuid}, but I can only find the conversation_id.

I've tried using the conversation_id in place of the UUID, but it doesn't work. Can anyone explain where I can find the UUID of the conversation or how I can generate a CSAT URL based on the conversation details?

Thank you very much for your help!

**Comment by sojan-official** [comment, marked as answer]
The CSAT URL needs the conversation `uuid`, not the display ID / conversation ID.

You can get it from the conversation API response. The conversation payload includes both:

```json
{
  "id": 123,
  "uuid": "conversation-uuid-here"
}
```

For the user-facing survey page, build the URL like this:

```text
https://your-chatwoot-domain.com/survey/responses/{conversation_uuid}
```

The public API endpoint behind this uses the same UUID:

```text
/public/api/v1/csat_survey/{conversation_uuid}
```

A couple of notes:

- The survey lookup is done by conversation `uuid`.
- The conversation should already have a CSAT survey message generated for it; otherwise the survey endpoint will not find a CSAT message to update.
